zerver/lib/test_helpers.py

```diff
@@ -726,10 +729,10 @@ def verify_serialize(
 
 def patch_queue_publish(
     method_to_patch: str,
-) -> Callable[[Callable[..., None]], Callable[..., None]]:
-    def inner(func: Callable[..., None]) -> Callable[..., None]:
+) -> Callable[[Callable[ParamT, None]], Callable[ParamT, None]]:
+    def inner(func: Callable[ParamT, None]) -> Callable[ParamT, None]:
         @wraps(func)
-        def _wrapped(*args: object, **kwargs: object) -> None:
+        def _wrapped(*args: ParamT.args, **kwargs: ParamT.kwargs) -> None:
             with mock_queue_publish(method_to_patch) as m:
                 func(*args, m, **kwargs)
```

Comment:
Hmm, this shouldn’t type check (even with `m: Any`), because we’re passing more arguments to `func` than expected by the signature. The signature should have been `func: Callable[Concatenate[ParamT, mock.MagicMock], None]`, except that isn’t allowed: `Concatenate` requires the `ParamSpec` to be at the end.

Probably the only real solution is to get rid of `patch_queue_publish` and just accept that we need to indent an extra level for `with mock_queue_publish(…)`.

Reply:
I see. Since `patch_queue_publish` is only used for methods in `zerver.tests.test_service_bot_system.TestServiceBotEventTriggers`, we can properly type this without `ParamSpec`.